### ISSUE TITLE
Documentation

### ISSUE THREAD

**Issue by lashus** [Mailtrain-org/mailtrain#301]
Tl;dr
I'd like to contribute by putting up together a documentation. Need help to build-up some basic plan.
---

Hello,

This project looks great and have potential although lack of documentation is a serious con - I'd like to help with that. Altough as I don't have much experience with your software/code (just diving into it) - it would be nice if we could put a "plan" for it.

@andris9 - as you're the top contributor I'd like to ask you for opinion.

IMO it should contain:

1. Getting started
2. Installation
3. Configuration Reference
4. Providers
5. Lists
    1. List managements
    2. List segmentation
6. Templates
7. Campaigns
8. Automation
9. Webhooks (plus instructions)
10. Advanced

**Comment by andris9** [comment]
Great idea! Do you have any suggestions how to present the docs as well? Use the Github Wiki? Or set up some kind of CMS like wordpress or Github pages that would be available from, for example, docs.mailtrain.org? I don't have a strong preference myself.

**Comment by lashus** [comment]
Well, i'm open to suggestions as well. To be honest - I was thinking about Github pages myself - it's well integrated with github itself, fairly easy to use and looking nice :)

**Comment by andris9** [comment]
Github pages sound great. I set up a /docs folder for the documentation and pointed it to http://docs.mailtrain.org/

**Comment by sommarnatt** [comment]
Better docs is a great idea!

Is there anyway to ask questions except using issues?

Been having a great time playing around with Mailtrain, but I've got some questions and need to know if I should "try" and fix them and add a merge request or if it's by design.

- Duplicate custom field names/merge tags allowed. DB has a multicolumn unique identifier, but it seems to use the wrong column for it. Since there's no "get custom fields" API endpoint, it's easy to end up with alot of dupes.
- Allow to set a campaign as "web viewable" by anonymous users. So we can post it to social media. If we use a dummy user to do that then anyone can unsubscribe it.

**Comment by lashus** [comment]
@andris9 Thanks for setting this up :) I hope to finish some basics before monday - after that i'm on vacations and won't spend much time in here.

@sommarnatt I guess PRs are always welcomed on open source projects :) If you want to ask specific questions please create separate issues.

**Comment by apertureless** [comment]
GitBook + GitHub Pages works pretty well. So you can write everything in markdown and it's easy to maintain.

**Comment by christianllv** [comment]
If you need I can translate the docs to spanish :)

**Comment by lashus** [comment]
@christianllv Great to hear that. I'm back from my vacations and will start making drafts soon :)

**Comment by programster** [comment]
I think this is a great idea and would love to provide any assistance I can with using Docker under the installation section. I think the first thing we need to do though is move 90% of the content from the README page to the docs area. Hopefully the Readme can be a nice landing page with the bare minimum to know what the project is and how to quickly deploy to try it out, before leading them into the documentation area for more information about all the little things like DKIM etc.

**Comment by lashus** [comment]
@programster thanks for insight :) but i think that's the idea that the documentation should have all the things that are currnetly scattered in various places. We will be using github pages for this purpose :) Hopefully by the end of the week I'll commit first version - sorry for this delay :/

**Comment by christianllv** [comment]
@lashus I'm wait for the first commit, and maybe I can translate to spanish and push commit too.

**Comment by programster** [comment]
All I did in my pull request is copy the README file as is into the docs area so it should have all the information that is _currently_ available there, before then slimming down the frontpage README. I saw this as a quick fix for a first iteration but will understand if you want to wait until you are 100% happy with a much more complete solution.
